Домашнее задание Модуль №10

Ulyanov-VN opened the pull request.

Ulyanov-VN committed:
Домашнее задание Модуль №10

ArturNurtdinov commented on app/build.gradle.kts:
```diff
@@ -37,6 +40,12 @@ android {
 
 dependencies {
 
+    implementation ("com.google.android.material:material:1.6.1")
+    implementation ("com.github.bumptech.glide:glide:4.14.2")
+    annotationProcessor ("com.github.bumptech.glide:compiler:4.14.2")
```
Все новые зависимости стоит делать через version catalog механизм - файл `libs.versions.toml`, чтобы все внешние зависимости проекта находились в одном месте - так удобнее будет с ними работать, а также все модули проекта будут зависеть от одних и тех же версий сторонних зависимостей

ArturNurtdinov commented on app/src/main/AndroidManifest.xml:
```diff
@@ -28,16 +29,16 @@
             android:name=".MediaActivity"
             android:exported="false" />
-
+        <!--        <activity
```
Закоментированный код лучше сразу удалять. Хранить закоментированный код в проекте - плохая практика, в случае чего этот код можно восстановить с помощью истории гита

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/TrackAdapter.kt:
```diff
@@ -0,0 +1,40 @@
+
+    override fun getItemCount(): Int = tracks.size
+
+    fun filter(query: String) {
```
Логику фильтрации сейчас можно удалить, т.к. уже в след спринте будет реализован настоящий поиск

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/TrackViewHolder.kt:
```diff
@@ -0,0 +1,37 @@
+                .placeholder(R.drawable.placeholder)
+                .error(R.drawable.placeholder) // Добавляем обработку ошибок
+                .centerCrop()
+                .transform(RoundedCorners(16))
```
В конструктор `RoundedCorners` нужно передавать кол-во пикселей. Это значит, что предварительно число, передаваемое в конструктор, нужно сконвертировать в пиксели, если мы хотим добиться одинакового скругления на девайсах с разной плотностью экрана

ArturNurtdinov reviewed (commented):
1. Плейсхолдер обложки не соответствует макетам:
![image](https://github.com/user-attachments/assets/91e10194-4c80-4f74-a673-ce9fc75ac0da)
Макет:
![image](https://github.com/user-attachments/assets/bf86fa9f-16c1-4d91-9136-e9126e40c461)
2. Отступ между текстом и обложкой не соответствует макетам, по макетам 8dp:
![image](https://github.com/user-attachments/assets/0290b99d-3b57-49f5-a448-936c61743f64)
Макет:
![image](https://github.com/user-attachments/assets/8a15a8c8-92bd-4bd9-93d3-a8533d7ea6f8)

Ulyanov-VN committed:
Домашнее задание Модуль №10 - устранение замечаний

1. Плейсхолдер обложки сделаны по макету для темной темы и светлой темы. Так же дабавлены строчки по очистке кеша, чтобы при отключении интернета отображались drawable.placeholder_dark/drawable.placeholder_light

2. Исправлен отступ между текстом и обложкой на 8dp.

ArturNurtdinov commented on app/src/main/res/drawable/placeholder_dark.xml:
Для тёмной и светлой темы плейсхолдер не отличается, поэтому можно не делить на dark и light
